include/deal.II/numerics/vector_tools.h

```diff
@@ -24,8 +24,14 @@
 #include <deal.II/base/std_cxx11/function.h>
 #include <deal.II/dofs/function_map.h>
 #include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
 #include <deal.II/hp/dof_handler.h>
 #include <deal.II/hp/mapping_collection.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/matrix_free/matrix_free.h>
```

Comment:
Originally, this was needed for the implementation of `project_generic` in `vector_tools.h`. Since we know at least some use cases, it should be fine to move it to `vector_tools.templates.h` which is done now.